# Homework 4 - pandas Arielle Pastore  
### due 3/14/2022   

Use this notebook and prompts to complete the homework. Throughout there will be hints and some code provided  

### Things you will need:  
- Install os, NumPy, pandas  
- states_covid.csv  
- Bloom_etal_2018_Reduced_Dataset.csv  
- logfiles.tgz (or some other multiple file dataset)  

*NOTE*: Make sure your PATH is correct  

**import packages & check required datasets**   

In [95]:
import os
import numpy as np
import pandas as pd

In [96]:
pwd

'/Users/ariellejewelpastore/Desktop/hw4/Part.3.PythonProgramming/Pandas'

In [97]:
PATH = '/Users/ariellejewelpastore/Desktop/hw4/Part.3.PythonProgramming/Pandas' #CHANGED

In [98]:
assert os.path.exists(os.path.join(PATH,'states_covid.csv')), 'states_covid.csv does not exist' 

In [99]:
assert os.path.exists(os.path.join(PATH,'Bloom_etal_2018_Reduced_Dataset.csv')), 'Bloom_etal_2018_Reduced_Dataset.csv does not exist'

## Task 1 - DataFrame manipulation  

Using **states_covid.csv**, we are going to read the data in as a DataFrame to manipulate, subset, and filter in various ways.   

### Task 1.1 

Read in states_covid.csv with date as a "date" dtype, and only columns consisting of the hospitalization (4 col), ICU (2 col), and Ventilators (2 col)  

In [112]:
state_covid_sub_df = pd.read_csv('states_covid.csv',usecols=['date','state','hospitalized','hospitalizedCumulative','hospitalizedCurrently','hospitalizedIncrease','inIcuCumulative','inIcuCurrently','onVentilatorCumulative','onVentilatorCurrently'],parse_dates=['date'],infer_datetime_format=True)
state_covid_sub_df.head()

date state  hospitalized  hospitalizedCumulative  \
0 2021-02-23    AK        1260.0                  1260.0   
1 2021-02-23    AL       45250.0                 45250.0   
2 2021-02-23    AR       14617.0                 14617.0   
3 2021-02-23    AS           NaN                     NaN   
4 2021-02-23    AZ       57072.0                 57072.0   

   hospitalizedCurrently  hospitalizedIncrease  inIcuCumulative  \
0                   38.0                     9              NaN   
1                  762.0                   122           2632.0   
2                  545.0                    47              NaN   
3                    NaN                     0              NaN   
4                 1515.0                    78              NaN   

   inIcuCurrently  onVentilatorCumulative  onVentilatorCurrently  
0             NaN                     NaN                    5.0  
1             NaN                  1497.0                    NaN  
2           204.0                  1505.0                   99.0  
3             NaN                     NaN                    NaN  
4           447.0                     NaN                  266.0

### Task 1.2 

For each of the following catergories: *currently* hospitalized, *currently* in the ICU, and *currently* on ventilation...  
Find the 5 states with the greatest numbers in each catergory and list them in order.      
*hint*: sort_values, unique  

In [113]:
#create subset with state
icu_state_df = state_covid_sub_df[['date','state','inIcuCurrently']]
icu_state_df.head()

date state  inIcuCurrently
0 2021-02-23    AK             NaN
1 2021-02-23    AL             NaN
2 2021-02-23    AR           204.0
3 2021-02-23    AS             NaN
4 2021-02-23    AZ           447.0

In [117]:
#make NAs zero
icu_state_df['inIcuCurrently'] = icu_state_df['inIcuCurrently'].fillna(0)
print(icu_state_df)

# set to descending order and only include unique values
icu_state_df.sort_values('inIcuCurrently',ascending=False,inplace=True)
print(icu_state_df)

            date state  inIcuCurrently
0     2021-02-23    AK             0.0
1     2021-02-23    AL             0.0
2     2021-02-23    AR           204.0
3     2021-02-23    AS             0.0
4     2021-02-23    AZ           447.0
...          ...   ...             ...
20103 2020-01-17    WA             0.0
20104 2020-01-16    WA             0.0
20105 2020-01-15    WA             0.0
20106 2020-01-14    WA             0.0
20107 2020-01-13    WA             0.0

[20108 rows x 3 columns]
            date state  inIcuCurrently
17677 2020-04-14    NY          5225.0
17621 2020-04-15    NY          5205.0
17789 2020-04-12    NY          5198.0
17733 2020-04-13    NY          5156.0
17565 2020-04-16    NY          5071.0
...          ...   ...             ...
11511 2020-08-02    ND             0.0
11512 2020-08-02    NE             0.0
11513 2020-08-02    NH             0.0
11515 2020-08-02    NM             0.0
20107 2020-01-13    WA             0.0

[20108 rows x 3 columns]


/var/folders/3l/mlkvk65x2lvfqppw7l4kthqm0000gn/T/ipykernel_15294/662628976.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  icu_state_df['inIcuCurrently'] = icu_state_df['inIcuCurrently'].fillna(0)
/var/folders/3l/mlkvk65x2lvfqppw7l4kthqm0000gn/T/ipykernel_15294/662628976.py:6: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  icu_state_df.sort_values('inIcuCurrently',ascending=False,inplace=True)


In [120]:
icu_state_df.drop_duplicates(subset='state', keep="first")

date state  inIcuCurrently
17677 2020-04-14    NY          5225.0
2413  2021-01-11    CA          4971.0
1951  2021-01-20    TX          3686.0
17674 2020-04-14    NJ          2051.0
17888 2020-04-10    MI          1663.0
3958  2020-12-15    OH          1311.0
16816 2020-04-29    IL          1290.0
3961  2020-12-15    PA          1264.0
2356  2021-01-12    AZ          1183.0
16933 2020-04-27    MA          1089.0
4665  2020-12-02    IN          1001.0
2158  2021-01-16    NC           880.0
2734  2021-01-06    TN           831.0
3330  2020-12-26    MO           685.0
16206 2020-05-10    MD           611.0
2009  2021-01-19    VA           588.0
2259  2021-01-14    KY           548.0
3063  2020-12-31    OK           501.0
2116  2021-01-17    SC           495.0
3508  2020-12-23    NV           460.0
2410  2021-01-11    AR           460.0
5541  2020-11-17    WI           456.0
4729  2020-12-01    MN           394.0
2324  2021-01-13    MS           360.0
4274  2020-12-09    KS           301.0
4252  2020-12-10    WA           294.0
5502  2020-11-17    IA           288.0
2176  2021-01-16    UT           231.0
2686  2021-01-07    WV           219.0
4240  2020-12-10    OR           139.0
16192 2020-05-10    DC           130.0
3711  2020-12-19    ID           118.0
5138  2020-11-24    PR           111.0
4581  2020-12-04    SD           111.0
4901  2020-11-28    MT            99.0
16507 2020-05-05    RI            89.0
16048 2020-05-13    NE            81.0
1927  2021-01-20    ME            71.0
3761  2020-12-18    DE            68.0
8581  2020-09-23    HI            61.0
6247  2020-11-04    ND            54.0
6228  2020-11-04    GU            22.0
947   2021-02-07    VT            20.0
15910 2020-05-15    CO             0.0
15763 2020-05-18    MP             0.0
14916 2020-06-02    LA             0.0
15242 2020-05-27    FL             0.0
15235 2020-05-27    AS             0.0
15993 2020-05-14    NH             0.0
15058 2020-05-31    VI             0.0
15511 2020-05-23    WY             0.0
15848 2020-05-16    AK             0.0
15239 2020-05-27    CT             0.0
15849 2020-05-16    AL             0.0
15995 2020-05-14    NM             0.0
15411 2020-05-24    GA             0.0

The top 5 states are NY, CA, TX, NJ, and MI.

### Task 1.3 

Find the date in which each state crossed 1000 cumulative hospitilized covid patients

In [148]:
cum_hos_df = state_covid_sub_df[['date','state','hospitalizedCumulative']]
cum_hos_df

date state  hospitalizedCumulative
0     2021-02-23    AK                  1260.0
1     2021-02-23    AL                 45250.0
2     2021-02-23    AR                 14617.0
3     2021-02-23    AS                     NaN
4     2021-02-23    AZ                 57072.0
...          ...   ...                     ...
20103 2020-01-17    WA                     NaN
20104 2020-01-16    WA                     NaN
20105 2020-01-15    WA                     NaN
20106 2020-01-14    WA                     NaN
20107 2020-01-13    WA                     NaN

[20108 rows x 3 columns]

In [149]:
cum_hos_df['hospitalizedCumulative'] = cum_hos_df['hospitalizedCumulative'].fillna(0)
print(cum_hos_df)

            date state  hospitalizedCumulative
0     2021-02-23    AK                  1260.0
1     2021-02-23    AL                 45250.0
2     2021-02-23    AR                 14617.0
3     2021-02-23    AS                     0.0
4     2021-02-23    AZ                 57072.0
...          ...   ...                     ...
20103 2020-01-17    WA                     0.0
20104 2020-01-16    WA                     0.0
20105 2020-01-15    WA                     0.0
20106 2020-01-14    WA                     0.0
20107 2020-01-13    WA                     0.0

[20108 rows x 3 columns]


/var/folders/3l/mlkvk65x2lvfqppw7l4kthqm0000gn/T/ipykernel_15294/2400372948.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  cum_hos_df['hospitalizedCumulative'] = cum_hos_df['hospitalizedCumulative'].fillna(0)


In [150]:
df = cum_hos_df.query("hospitalizedCumulative >= 1000")
df

date state  hospitalizedCumulative
0     2021-02-23    AK                  1260.0
1     2021-02-23    AL                 45250.0
2     2021-02-23    AR                 14617.0
4     2021-02-23    AZ                 57072.0
6     2021-02-23    CO                 23293.0
...          ...   ...                     ...
18797 2020-03-25    NY                  5126.0
18853 2020-03-24    NY                  4041.0
18909 2020-03-23    NY                  3125.0
18965 2020-03-22    NY                  2354.0
19021 2020-03-21    NY                  1531.0

[9308 rows x 3 columns]

In [151]:
df.sort_values('hospitalizedCumulative',ascending=True,inplace=True)
df

/var/folders/3l/mlkvk65x2lvfqppw7l4kthqm0000gn/T/ipykernel_15294/668318139.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.sort_values('hospitalizedCumulative',ascending=True,inplace=True)


date state  hospitalizedCumulative
3495  2020-12-23    ME                  1000.0
1713  2021-01-24    NH                  1000.0
16844 2020-04-29    SC                  1000.0
16788 2020-04-30    SC                  1000.0
14424 2020-06-11    NE                  1000.0
...          ...   ...                     ...
13701 2020-06-24    NY                 89995.0
7821  2020-10-07    NY                 89995.0
5749  2020-11-13    NY                 89995.0
7317  2020-10-16    NY                 89995.0
14485 2020-06-10    NY                 89995.0

[9308 rows x 3 columns]

In [153]:
df.drop_duplicates(subset='state', keep="first")

date state  hospitalizedCumulative
3495  2020-12-23    ME                  1000.0
1713  2021-01-24    NH                  1000.0
16844 2020-04-29    SC                  1000.0
14424 2020-06-11    NE                  1000.0
7253  2020-10-17    MT                  1001.0
7181  2020-10-18    HI                  1001.0
16996 2020-04-26    MS                  1001.0
3807  2020-12-18    WY                  1001.0
14879 2020-06-03    OK                  1003.0
14170 2020-06-15    AR                  1003.0
3136  2020-12-29    AK                  1004.0
18241 2020-04-04    PA                  1004.0
13648 2020-06-25    OR                  1006.0
10013 2020-08-29    SD                  1006.0
10991 2020-08-11    ID                  1006.0
18238 2020-04-04    OH                  1006.0
17435 2020-04-18    KY                  1008.0
16689 2020-05-01    AL                  1008.0
14074 2020-06-17    KS                  1011.0
14328 2020-06-13    UT                  1012.0
16846 2020-04-29    TN                  1013.0
16619 2020-05-03    RI                  1013.0
7759  2020-10-08    ND                  1016.0
17812 2020-04-11    AZ                  1037.0
16769 2020-04-30    MN                  1044.0
17577 2020-04-16    VA                  1048.0
17693 2020-04-14    WI                  1049.0
18323 2020-04-02    GA                  1056.0
18110 2020-04-06    MD                  1059.0
18038 2020-04-07    CO                  1079.0
18322 2020-04-02    FL                  1123.0
15491 2020-05-23    NM                  1139.0
18221 2020-04-04    MA                  1370.0
19021 2020-03-21    NY                  1531.0
15620 2020-05-21    WA                  3125.0
15921 2020-05-15    IN                  4389.0
16695 2020-05-01    CT                  7758.0
15322 2020-05-26    NJ                 16373.0

## Task 2 - DataFrame summarizing  

Using **Bloom_etal_2018_Reduced_Dataset.csv**, we are going to do more dataframe manipulation and subsetting and summarizing    

### Task 2.1 

Read in Bloom_etal_2018_Reduced_Dataset.csv and create two new columns ('genus','species') that consists of the column *taxa* split at the underscore. Print out the head of this new dataframe and the number of unique genera**     

*hint:* pd.str.split(,expand=True)  

for example:  

| taxa | genus | species   
| :------ | :-- | :---   
| Alosa_alabamae | Alosa | alabamae  

In [169]:
df = pd.read_csv('Bloom_etal_2018_Reduced_Dataset.csv',usecols=['taxa'])
df

taxa
0                Alosa_alabamae
1                   Alosa_alosa
2                  Alosa_fallax
3               Alosa_mediocris
4          Alosa_pseudoharengus
5             Alosa_sapidissima
6      Amazonsprattus_scintilla
7              Anchoa_chamensis
8                 Anchoa_cubana
9          Anchoa_delicatissima
10          Anchoa_lamprotaenia
11              Anchoa_lyolepis
12                 Anchoa_parva
13            Anchoa_schofieldi
14        Anchovia_surinamensis
15          Anchoviella_balboae
16     Anchoviella_brevirostris
17           Anchoviella_jamesi
18   Anchoviella_lepidentostole
19       Anchoviella_manamensis
20       Anodontostoma_chacunda
21       Cetengraulis_edentulus
22  Chirocentrodon_bleekerianus
23      Clupeichthys_perakensis
24        Clupeoides_borneensis
25                 Coilia_nasus
26          Ehirava_fluviatilis
27        Encrasicholina_devisi
28          Engraulis_eurystole
29          Escualosa_thoracata
30          Ethmalosa_fimbriata
31      Gilchristella_aestuaria
32                  Hilsa_kelee
33         Hyperlophus_vittatus
34       Jenkinsia_lamprotaenia
35         Microthrissa_congica
36            Pellona_harroweri
37          Pellonula_leonensis
38              Pellonula_vorax
39        Potamalosa_richmondia
40               Ramnogaster_sp
41      Rhinosardinia_bahiensis
42           Sauvagella_robusta
43      Sierrathrissa_leonensis
44    Spratelloides_delicatulus
45           Stolephorus_waitei
46      Stolothrissa_tanganicae
47      Sundasalanx_mekongensis
48             Tenualosa_ilisha
49        Thryssa_cf_dussumieri

In [179]:
new = df["taxa"].str.split("_", n = 1, expand = True)
df["genus"]= new[0]
df["species"]= new[1]
df

taxa           genus         species
0                Alosa_alabamae           Alosa        alabamae
1                   Alosa_alosa           Alosa           alosa
2                  Alosa_fallax           Alosa          fallax
3               Alosa_mediocris           Alosa       mediocris
4          Alosa_pseudoharengus           Alosa  pseudoharengus
5             Alosa_sapidissima           Alosa     sapidissima
6      Amazonsprattus_scintilla  Amazonsprattus       scintilla
7              Anchoa_chamensis          Anchoa       chamensis
8                 Anchoa_cubana          Anchoa          cubana
9          Anchoa_delicatissima          Anchoa   delicatissima
10          Anchoa_lamprotaenia          Anchoa    lamprotaenia
11              Anchoa_lyolepis          Anchoa        lyolepis
12                 Anchoa_parva          Anchoa           parva
13            Anchoa_schofieldi          Anchoa      schofieldi
14        Anchovia_surinamensis        Anchovia    surinamensis
15          Anchoviella_balboae     Anchoviella         balboae
16     Anchoviella_brevirostris     Anchoviella    brevirostris
17           Anchoviella_jamesi     Anchoviella          jamesi
18   Anchoviella_lepidentostole     Anchoviella  lepidentostole
19       Anchoviella_manamensis     Anchoviella      manamensis
20       Anodontostoma_chacunda   Anodontostoma        chacunda
21       Cetengraulis_edentulus    Cetengraulis       edentulus
22  Chirocentrodon_bleekerianus  Chirocentrodon    bleekerianus
23      Clupeichthys_perakensis    Clupeichthys      perakensis
24        Clupeoides_borneensis      Clupeoides      borneensis
25                 Coilia_nasus          Coilia           nasus
26          Ehirava_fluviatilis         Ehirava     fluviatilis
27        Encrasicholina_devisi  Encrasicholina          devisi
28          Engraulis_eurystole       Engraulis       eurystole
29          Escualosa_thoracata       Escualosa       thoracata
30          Ethmalosa_fimbriata       Ethmalosa       fimbriata
31      Gilchristella_aestuaria   Gilchristella       aestuaria
32                  Hilsa_kelee           Hilsa           kelee
33         Hyperlophus_vittatus     Hyperlophus        vittatus
34       Jenkinsia_lamprotaenia       Jenkinsia    lamprotaenia
35         Microthrissa_congica    Microthrissa         congica
36            Pellona_harroweri         Pellona       harroweri
37          Pellonula_leonensis       Pellonula       leonensis
38              Pellonula_vorax       Pellonula           vorax
39        Potamalosa_richmondia      Potamalosa      richmondia
40               Ramnogaster_sp     Ramnogaster              sp
41      Rhinosardinia_bahiensis   Rhinosardinia       bahiensis
42           Sauvagella_robusta      Sauvagella         robusta
43      Sierrathrissa_leonensis   Sierrathrissa       leonensis
44    Spratelloides_delicatulus   Spratelloides     delicatulus
45           Stolephorus_waitei     Stolephorus          waitei
46      Stolothrissa_tanganicae    Stolothrissa      tanganicae
47      Sundasalanx_mekongensis     Sundasalanx     mekongensis
48             Tenualosa_ilisha       Tenualosa          ilisha
49        Thryssa_cf_dussumieri         Thryssa   cf_dussumieri

### Task 2.2 

Create a new dataframe with the mean *logbodysize* and *trophicposition* of each genera. Sort this data frame by the largest body size. Print the head of this dataframe.

In [195]:
df2 = pd.read_csv('Bloom_etal_2018_Reduced_Dataset.csv',usecols=['taxa','logbodysize','trophic_position'])
new2 = df2["taxa"].str.split("_", n = 1, expand = True)
df2["genus"]= new2[0]
df2. __delitem__('taxa')
df2 = df2[['genus','logbodysize', 'trophic_position']]
df2

genus  logbodysize  trophic_position
0            Alosa     1.707570          0.431364
1            Alosa     1.778151          0.556303
2            Alosa     1.778151          0.556303
3            Alosa     1.778151          0.612784
4            Alosa     1.602060          0.544068
5            Alosa     1.790285          0.544068
6   Amazonsprattus     0.301030          0.531479
7           Anchoa     0.778151          0.518514
8           Anchoa     0.845098          0.518514
9           Anchoa     0.903090          0.518514
10          Anchoa     0.963788          0.531479
11          Anchoa     0.845098          0.531479
12          Anchoa     0.778151          0.518514
13          Anchoa     0.792392          0.518514
14        Anchovia     1.093422          0.477121
15     Anchoviella     0.857332          0.505150
16     Anchoviella     0.857332          0.505150
17     Anchoviella     0.698970          0.491362
18     Anchoviella     1.064458          0.491362
19     Anchoviella     0.397940          0.491362
20   Anodontostoma     1.342423          0.447158
21    Cetengraulis     1.113943          0.322219
22  Chirocentrodon     0.954243          0.518514
23    Clupeichthys     0.602060          0.477121
24      Clupeoides     0.903090          0.491362
25          Coilia     1.544068          0.477121
26         Ehirava     0.698970          0.491362
27  Encrasicholina     0.903090          0.518514
28       Engraulis     1.190332          0.462398
29       Escualosa     1.000000          0.505150
30       Ethmalosa     1.544068          0.397940
31   Gilchristella     1.000000          0.491362
32           Hilsa     1.387390          0.462398
33     Hyperlophus     1.000000          0.531479
34       Jenkinsia     0.838849          0.518514
35    Microthrissa     0.778151          0.491362
36         Pellona     1.113943          0.623249
37       Pellonula     0.929419          0.518514
38       Pellonula     1.079181          0.491362
39      Potamalosa     1.505150          0.518514
40     Ramnogaster     0.903090          0.531479
41   Rhinosardinia     0.903090          0.531479
42      Sauvagella     0.672098          0.491362
43   Sierrathrissa     0.477121          0.491362
44   Spratelloides     0.845098          0.491362
45     Stolephorus     0.973128          0.518514
46    Stolothrissa     1.000000          0.431364
47     Sundasalanx     0.447158          0.491362
48       Tenualosa     1.778151          0.462398
49         Thryssa     1.041393          0.447158

Which genera is the smallest and largest?

In [199]:
df2.sort_values('logbodysize',ascending=True,inplace=True)
print(df2)


             genus  logbodysize  trophic_position
6   Amazonsprattus     0.301030          0.531479
19     Anchoviella     0.397940          0.491362
47     Sundasalanx     0.447158          0.491362
43   Sierrathrissa     0.477121          0.491362
23    Clupeichthys     0.602060          0.477121
42      Sauvagella     0.672098          0.491362
17     Anchoviella     0.698970          0.491362
26         Ehirava     0.698970          0.491362
7           Anchoa     0.778151          0.518514
35    Microthrissa     0.778151          0.491362
12          Anchoa     0.778151          0.518514
13          Anchoa     0.792392          0.518514
34       Jenkinsia     0.838849          0.518514
8           Anchoa     0.845098          0.518514
44   Spratelloides     0.845098          0.491362
11          Anchoa     0.845098          0.531479
16     Anchoviella     0.857332          0.505150
15     Anchoviella     0.857332          0.505150
40     Ramnogaster     0.903090          0.531479


the smallest is Amazonsprattus and the largest is Alosa

What is the trophic position of the smallest and largest?  

0.531479 and 0.544068, respectively

## Task 3 - Read in muliple files to a dictionary and make a DataFrame - **OPTIONAL/BONUS**  

### This is not something you are expected to do in this course, but just here to give you an idea of the things that you COULD do. Answers will be posted after due date.  


Using **logfiles**: we are going to do read in each file, get some data, append it to a dictionary to later make into a dataframe.     

**note:** *make sure to unzip logfiles*

In [20]:
cd $PATH

/Users/trevorfaske/Desktop/Classes/DataScience/pandas/Spring22


In [21]:
!tar -xzf logfiles.tgz #unzip logfiles

In [22]:
log_dir = os.path.join(PATH,'logfiles')
assert os.path.exists(log_dir), 'log_dir does not exist'

First step is to find the necessary files. The number of files in the log files is 36, make sure you have that many as well  

In [23]:
!ls -l logfiles/*txt | wc -l 

      36


In [24]:
logfiles = !find $log_dir -name '*txt' #unix command to find files in log_dir directory
logfiles = [os.path.abspath(x) for x in logfiles] #this finds the full path to the file
print(logfiles[1])

/Users/trevorfaske/g/Classes/DataScience/pandas/Spring22/logfiles/1901302121_H8_S_14.txt.txt


In [25]:
assert(len(logfiles)==36), 'Do not have correct number of logfiles'

### Getting a little tricky here  

Read in each of the logfiles, for each file extract:  
- minimum temperature  
- maximum temperature  
- date of minimum temp   
- date of maximum temp   
- mean temp for each file.   

This data should all be appended for a dictionary within a for loop:    
Key should be the file name without the path or .txt extension  
Values should be (minTemp,maxTemp,minDate,maxDate,meanTemp)

I recommend making this work for one file first, then putting the rest in a for loop to do the rest.  

Below is an example of how to read in one file

*hint:* do not read date in as date object

In [41]:
### set up data frame as you read in each file
infile = pd.read_csv(logfiles[0],sep='\t',engine='python')
infile.columns = ['Index','Date','Time','Temp','Type']
infile = infile[['Date','Time','Temp']]
infile.head()

Date         Time  Temp
0  9/16/2013   8:00:00 AM  47.9
1  9/16/2013   8:35:00 AM  48.2
2  9/16/2013   9:10:00 AM  48.7
3  9/16/2013   9:45:00 AM  49.4
4  9/16/2013  10:20:00 AM  50.3

Do everything in steps, make sure it works. Calculate summaries with this one infile:

In [26]:
minTemp =   
maxTemp =  
minDate = infile['Date'][infile['Temp'] == infile['Temp'].min()].unique()[0] #use this for minDate 
maxDate = infile['Date'][infile['Temp'] == infile['Temp'].max()].unique()[0] #use this for maxDate 
meanTemp = 

SyntaxError: invalid syntax (<ipython-input-26-d888355c2242>, line 1)

To get you started, I suggest writing some dummy code in plain words to help outline your for loop:  

In [ ]:
logfiles_dict = {}  
for f in logfiles:  
    #do something   
    #do not read date in as date object  
    #do more something   
    #do other stuff  
    #make print statements EVERYWHERE  
    #append to dict  
    #blahbahblah

Then do the real code below here. You don't need to turnin your thoughts. Just put it in there as a help reminder. Most people all still do this, no matter how advanced they are

**Once you have created a DataFrame with all the logfiles, print the head and save it to an outfile using pd.to_csv() as logfiles_df.csv** 

#### Here is an example of the final product

In [32]:
logfiles_df.head()

minTemp  maxTemp     minDate    maxDate   meanTemp
1901302235_H6_S_14      21.3     65.8   12/6/2013  7/13/2014  35.726191
1901302121_H8_S_14      17.9     73.4  10/28/2013  7/13/2014  34.698866
1901302217_H18_D_14     21.0     82.4  10/28/2013  7/26/2014  33.334564
1901302110_H13_S_14     22.8     60.9   10/4/2013  7/13/2014  32.240032
1901302146_H14_D_14     24.6     66.1   1/21/2014  7/13/2014  34.242075